src/core/stac/qgsstacitem.cpp

```diff
@@ -177,3 +177,61 @@ QgsDateTimeRange QgsStacItem::dateTimeRange() const
   const QDateTime end = QDateTime::fromString( mProperties.value( QStringLiteral( "end_datetime" ), QStringLiteral( "null" ) ).toString() );
   return QgsDateTimeRange( start, end );
 }
+
+QString QgsStacItem::title() const
+{
+  return mProperties.value( QStringLiteral( "title" ) ).toString();
+}
+
+QString QgsStacItem::description() const
+{
+  return mProperties.value( QStringLiteral( "description" ) ).toString();
+}
+
+QgsMimeDataUtils::UriList QgsStacItem::uris() const
+{
+  QgsMimeDataUtils::UriList uris;
+  for ( auto it = mAssets.constBegin(); it != mAssets.constEnd(); ++it )
+  {
+    QgsMimeDataUtils::Uri uri;
+    QUrl url( it->href() );
+    if ( url.isLocalFile() )
+    {
+      uri.uri = it->href();
+    }
+    else if ( it->mediaType() == QLatin1String( "image/tiff; application=geotiff; profile=cloud-optimized" ) ||
+              it->mediaType() == QLatin1String( "image/vnd.stac.geotiff; cloud-optimized=true" ) )
+    {
+      uri.layerType = QStringLiteral( "raster" );
+      uri.providerKey = QStringLiteral( "gdal" );
+      if ( it->href().startsWith( QLatin1String( "http" ), Qt::CaseInsensitive ) ||
+           it->href().startsWith( QLatin1String( "ftp" ), Qt::CaseInsensitive ) )
+      {
+        uri.uri = QStringLiteral( "/vsicurl/%1" ).arg( it->href() );
+      }
+      else if ( it->href().startsWith( QLatin1String( "s3://" ), Qt::CaseInsensitive ) )
+      {
+        uri.uri = QStringLiteral( "/vsis3/%1" ).arg( it->href().mid( 5 ) );
```

Comment:
Does this actually work? I suspect we'd generally need to also set the credentialOptions in the uri, eg to add the AWS_NO_SIGN_REQUEST=Yes option. I'm not aware of any s3 connections which will work without ANY credential options set.

I think in practice we'll need to use the storage extension to determine appropriate credentials -- see https://github.com/stac-extensions/storage

Reply:
> Users will need to set up their gdal env variables or ~/.aws/config files and their layers will just work!

Right -- but since https://github.com/qgis/QGIS/pull/57826 we can do better, because we could just use credentialOptions in the layer URI and set this for users :smile: Then things really will "just work", no config file messing around required.

> I also haven't seen the storage stac-extension used on any of the servers I've used for testing, so I'd be reluctant to depend on it.

Try https://earth-search.aws.element84.com/v1/collections/ -- that one uses the storage extension to specify:

```
"storage:platform": [
          "AWS"
        ],
        "storage:region": [
          "eu-central-1"
        ],
        "storage:requester_pays": [false]
```

Reply:
I've checked the extension, we could indeed handle the cases where `requester_pays=false` and use `AWS_NO_SIGN_REQUEST=YES` (shouldn't this be default when no other signing info are provided?) and maybe also use the region in order to translate the `s3://` url to `https://` for downloading assets.
As for CredentialOptions, AFAICT they are gdal specific. I'd prefer to have something more generic (like the config files) that can also work outside gdal for downloading assets.
We could use the existing `QgsAuthAwsS3Method`  and in the STAC connection we could define a list of authCfgs which should be used for downloading assets, as they will probably be different from the catalog's authcfg.
Then the AuthManager would have a method `QStringList getAvailableAuthCfgsForThisLink( url )` and we could decide which one to use for downloading the files.

But as I said before, all this handling of s3 links is well outside the scope of this PR

Reply:
> I've checked the extension, we could indeed handle the cases where requester_pays=false and use AWS_NO_SIGN_REQUEST=YES (shouldn't this be default when no other signing info are provided?) 

I'd say that's a safe call to make, but @rouault will likely know better

> maybe also use the region in order to translate the s3:// url to https:// for downloading assets.

I'd suggest just directly mapping that to the "AWS_DEFAULT_REGION" credential option, and stick with vsis3 and let GDAL handle the rest.

> As for CredentialOptions, AFAICT they are gdal specific. I'd prefer to have something more generic (like the config files) that can also work outside gdal for downloading assets.

They are GDAL/OGR provider specific, but we should definitely use them when loading GDAL/OGR layers from the collections. That we we pass off all the handling of this and optimisations and such to GDAL, and it's the BEST place to do that :laughing: 

> We could use the existing QgsAuthAwsS3Method and in the STAC connection we could define a list of authCfgs which should be used for downloading assets, as they will probably be different from the catalog's authcfg.
Then the AuthManager would have a method QStringList getAvailableAuthCfgsForThisLink( url ) and we could decide which one to use for downloading the files.

I'm -1 to using auth methods for this (in general). They don't work at all if you share projects or layer configuration, and for the simple cases of AWS_NO_SIGN_REQUEST/AWS_DEFAULT_REGION there's nothing sensitive in the configuration which requires use of the secure auth framework.

I guess the question is whether or not there's ever going to be a stac collection where the items require DIFFERENT authentication vs the collection itself. If not, then could just handle the auth configuration in the stac connection and just copy the authcfg key from the connection to the loaded layers. 

> But as I said before, all this handling of s3 links is well outside the scope of this PR

That's fair enough -- but I'd go back to suggesting that we then ALWAYS add the "AWS_NO_SIGN_REQUEST=Yes" credential option, and just hope that it works :laughing:

Reply:
> I'd say that's a safe call to make, but @rouault will likely know better

I'm not sure. aws cli requires an explicit --no-sign-request.

> or not there's ever going to be a stac collection where the items require DIFFERENT authentication vs the collection itself

https://github.com/OSGeo/gdal/issues/11317 / https://github.com/OSGeo/gdal/pull/11319 is an example where the authentication to the STAC API is different than the assets

Reply:
> > maybe also use the region in order to translate the s3:// url to https:// for downloading assets.
> 
> I'd suggest just directly mapping that to the "AWS_DEFAULT_REGION" credential option, and stick with vsis3 and let GDAL handle the rest.

Can we use vsis3 to download assets? Assets are not necessarily gdal supported formats!

> I'm -1 to using auth methods for this (in general). They don't work at all if you share projects or layer configuration

I don't mean for loading assets as layers, this should definitely be done using vsis3. My point is for downloading assets as files, which may also be arbitrary non-gdal formats.

> That's fair enough -- but I'd go back to suggesting that we then ALWAYS add the "AWS_NO_SIGN_REQUEST=Yes" credential option, and just hope that it works 😆

Will it work if the link actually requires signing and the user has properly setup config files?